data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java

```diff
@@ -133,28 +135,39 @@ CombinedChainDataClient getCombinedChainDataClient() {
     }
     return combinedChainDataClient
         .getStateByBlockRoot(blockRoot)
-        .thenApply(state -> state.map(BeaconState::new))
-        .exceptionally(err -> Optional.empty());
+        .thenApply(state -> state.map(BeaconState::new));
   }
 
   public SafeFuture<Optional<BeaconState>> getStateAtSlot(UnsignedLong slot) {
-    if (!isStoreAvailable()) {
-      return SafeFuture.failedFuture(new ChainDataUnavailableException());
-    }
-    return combinedChainDataClient
-        .getStateAtSlot(slot)
-        .thenApply(state -> state.map(BeaconState::new))
-        .exceptionally(err -> Optional.empty());
+    return SafeFuture.of(
+        () -> {
+          if (!isStoreAvailable()) {
+            return SafeFuture.failedFuture(new ChainDataUnavailableException());
```

Comment:
nit: static import failedFuture?

Reply:
I tend to stay away from static imports for the most part 🤷‍♀